src/main/java/com/epam/reportportal/spock/ReportPortalSpockListener.java

```diff
@@ -74,6 +75,7 @@ public class ReportPortalSpockListener extends AbstractRunListener {
 					.orElseThrow(() -> new IllegalStateException("Unknown Spock version.")));
 
 	private final MemoizingSupplier<Launch> launch;
+	private final Queue<String> exceptionMessageQueue = new LinkedList<>();
```

Comment:
This won't work for concurrent execution. It will produce a mess of different errors from different tems.

Reply:
So, in that case probably it should be a synchronized Map, isn't it? Where key is a unique identifier and value is an error message. Tell me please, what better to use as a unique identifier?

Reply:
Approach was changed